### ISSUE TITLE
Undiscovered bad indentation in one of RuboCop's spec files

### ISSUE THREAD

**Issue by bquorning** [rubocop-hq/rubocop#5994]
I stumbled on a case of bad indentation in one of our specs:

https://github.com/rubocop-hq/rubocop/blob/c4e9a4f5f89bbd45b9687354179e211518690b2d/spec/rubocop/cop/style/parallel_assignment_spec.rb#L115-L126

I suspect it is related to the “concatenating strings over multiple lines using `\`” feature – which makes #4719 related too, I guess.

--------

## Expected behavior

RuboCop should have complained about bad indentation.

## Actual behavior

RuboCop is silent.

## Steps to reproduce the problem

In a checkout of the Rubocop source code:
```
git checkout c4e9a4f5f89bbd45b9687354179e211518690b2d
bundle exec rubocop spec/rubocop/cop/style/parallel_assignment_spec.rb
```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.57.2 (using Parser 2.5.1.0, running on ruby 2.5.1 x86_64-darwin15)
```

**Comment by bbatsov** [comment]
@bquorning That has been broken for a long time, but I never found the time to address it. I've seen this problem at several places in the codebase.

**Comment by Nondv** [comment]
I just looked into it.

Interesting thing is how parser count columns for parts of the string:
```ruby
  'abc' \
  'def' \
  'ghi'
```

```ruby
node.children.first.loc.column # ==> 2
node.children.last.loc.column # ==> 10
node.children.last.loc.column # ==> 18
```

So it seems like parser treats line break as a part of line since it is escaped via `\` which is logical (I guess).

We can work this out by using `#source_range` and so on, but since we cant (or can we?) say where newline char is I cant  find a way to implement this cop. One can write this literals different ways:

```ruby
'abc' \
'def'

'abc'\
'def'

'abc'      \
'def'

'or' 'even'   'like' \
'this'
```

I have never worked with rubocop and parser, so maybe I'm missing something.

@bbatsov

**Comment by bquorning** [comment]
Yes, I think `parser` (or `ast`?) is currently too limited to fix this issue.
